astropy/io/ascii/tests/test_c_reader.py

```diff
@@ -971,43 +979,45 @@ def test_data_out_of_range(parallel, reader):
     """
     # Python reader and strtod() are expected to return precise results
     rtol = 1.e-30
-    if reader > 1:
-        rtol = 1.e-15
-    # passing fast_reader dict with parametrize does not work!
-    if reader > 0:
-        fast_reader = {'parallel': parallel, 'use_fast_converter': reader > 1}
-    else:
-        fast_reader = False
+
+    # update fast_reader dict; pass only copies to avoid changing during read()!
+    if reader:
+        reader['parallel'] = parallel
+        if reader.get('use_fast_converter'):
+            rtol = 1.e-15
+
     if parallel:
-        if reader < 1:
+        if not reader:
             pytest.skip("Multiprocessing only available in fast reader")
         elif TRAVIS:
             pytest.xfail("Multiprocessing can sometimes fail on Travis CI")
 
     fields = [ '10.1E+199', '3.14e+313', '2048e+306', '0.6E-325', '-2.e345' ]
     values = np.array([ 1.01e200, np.inf, np.inf, 0.0, -np.inf ])
-    t = ascii.read(StringIO(' '.join(fields)), format='no_header', guess=False,
-                   fast_reader=fast_reader)
+    t = ascii.read(StringIO(' '.join(fields)), format='no_header',
+                   fast_reader=copy.deepcopy(reader))
```

Comment:
Then it is, I suppose - basically the same that made `guess`ing fail for the fast readers in the first place:
```
>>> fast_reader = {'parallel': False, 'use_fast_converter': True }
>>> fields = [ '.0101E202', '0.000000314E+314', '1777E+305', '-1799E+305', '0.2e-323' ]
>>> fast_reader
{'parallel': False, 'use_fast_converter': True}
>>> t = astropy.io.ascii.read(StringIO(' '.join(fields)), format='no_header', guess=False, fast_reader=fast_reader)
WARNING: OverflowError converting to FloatType in column col4, possibly resulting in degraded precision. [astropy.io.ascii.fastbasic]
WARNING: OverflowError converting to FloatType in column col5, possibly resulting in degraded precision. [astropy.io.ascii.fastbasic]
>>> fast_reader
{}
```